regions_www/m/pct_shiny_funs.R

```diff
@@ -30,7 +30,20 @@ normalise <- function(values, min = 0, max = 1){
 }
 
 
-## Create percentages such that rounded to 1DP if >0% and <0.5%.  [NB can be NaN if denominator zero]
+## Create percentages such that rounded to XDP if >0% and <0.5%.  [NB can be NaN if denominator zero]
+round_dp <- function(data, expression){
+  value <- round(expression)
+  for(i in 1:length(data)) {
+    if(!is.na(expression[i]) && !is.nan(expression[i]) && expression[i]>=0.05 && expression[i]<0.5) {
+      value[i] <- round(expression[i], 1)
+    }
+    if(!is.na(expression[i]) && !is.nan(expression[i]) && expression[i]>0 && expression[i]<0.05) {
+      value[i] <- round(expression[i], 2)
+    }
+    if (is.nan(value[i]) || is.na(value[i])) { value[i] <- "-" }
+  }
+  value
+}
```

Comment:
I would use `dplyr::recode()` to make it more concise - think that would work: http://dplyr.tidyverse.org/reference/recode.html

But happy to be a 'for the future' refactor. - maybe add a comment linking to this would be a good plan:

Reply:
I don't really think it reads clearly now (e.g. I don't see why `data` is being passed in).  I would like to tidy it before accepting this PR.

Reply:
good question re. passing data in, I agree that is not necessary.  what you wrote, Nikolai, looks much more elegant than what I had, so I've changed to that - plus also realised that the round_percent function was not really needed, could just be a special case of round_dp

Reply:
ah, ok, but I am now getting an error message after making this change: 

"Warning: Error in <-: NAs are not allowed in subscripted assignments"

I have reverted, maybe you could do and check the tidying, Nikolai? (or we leave for now)